File: src/app/Models/customer.ts
import { Address } from "./address";
import { Order } from "./order";
import { User } from "./user";

export class Customer extends User {
    productsPurchased: Order[];
    addresses: Address[];
    friendRequests: string[]; 
    friends: string[]; 
    friendProductRecommendations: string[];

    constructor(
        _id: string, 
        firstName: string,
        lastName: string, 
        email: string, 
        mobileNumber: string, 
        dateOfBirth: Date, 
        password: string,
        role: string,
        productsPurchased: Order[], 
        addresses: Address[], 
        friendRequests: string[], 
        friends: string[], 
        friendProductRecommendations: string[],
    ) {
        super(_id, firstName, lastName, email, password, mobileNumber, dateOfBirth, role);
        this.productsPurchased = productsPurchased;
        this.addresses = addresses;
        this.friendRequests = friendRequests;
        this.friends = friends;
        this.friendProductRecommendations = friendProductRecommendations;
    }
}

File: GoServer/Model/customer.go
package Model

type Customer struct {
	Id        						string      `json:"_id" bson:"_id"`
	FirstName 						string   	`json:"firstName" bson:"firstName" `
	LastName  						string   	`json:"lastName" bson:"lastName" `
	Email     						string   	`json:"email" bson:"email" `
	Password  						string   	`json:"password" bson:"password" `
	MobileNumber    				string   	`json:"mobileNumber" bson:"mobileNumber"`
	DateOfBirth     				string   	`json:"dateOfBirth" bson:"dateOfBirth"`
	Role      						string   	`json:"role" bson:"role"` 
	ProductsPurchased 				[]Order 	`json:"productsPurchased" bson:"productsPurchased"`
	Addresses 		 				[]Address 	`json:"addresses" bson:"addresses"`
	FriendRequests 					[]string 	`json:"friendRequests" bson:"friendRequests"`
	Friends 						[]string 	`json:"friends" bson:"friends"`
	FriendProductRecommendations 	[]string 	`json:"friendProductRecommendations" bson:"friendProductRecommendations"`
}